### ISSUE TITLE
Cache datafiles in local storage

### ISSUE THREAD

**Issue by ascholerChemeketa** [RunestoneInteractive/rs#423]
When an activecode accesses a datafile that is not on the page, the DB gets queried for it.

Once we get that datafile, it should maybe be cached into local storage and retrieved from there instead of going back to the DB for the file.

**Comment by moisedk** [comment]
There is no timestamp on the data files (I've looked in source_code table, and inspected the request's result in activecode.js)  to manage stale cache. I was going to add that to the source_code table, but I am not sure if that is necessary. Anyone has any suggestion?

**Comment by bnmnetp** [comment]
You would need to add the column to the `source_code` table.  But I'm really convinced that this is a big win.  Maybe @ascholerChemeketa has some use cases, but my own analysis of the log data would indicate that this is not a common enough use case to warrant caching.

**Comment by moisedk** [comment]
Did you mean you’re _not_ convinced it’s a big win, then? I was thinking it isn’t, too, but I wanted to make sure that it’s not a critical problem before I drop abandon it.

**Comment by ascholerChemeketa** [comment]
I just did a mini audit of network traffic on a book or two that use datafiles. The traffic involved doesn't seem like a big deal - with the caveat that all the files involved are relatively small. A couple K of text or images that are like 300x200 don't seem worth the complexity of caching. 

If an author did upload a multi MB text file or a giant image as a datafile, those datafiles could become an annoyance for students on slow connections. At one point in developing a book I did decide to intentionally trim down a datafile that was huge. But that was more because an Activecode wasn't a great place to work on a giant datafile. It was more about the experience of trying to run Skulpt interpreted Python on a ton of data and debug any output in the Activecode environment. Similarly, doing any complex work on a large image in Skulpt is REALLY slow.

SQLite DB files seem more reasonable to be large (Bikeshare used in Overview samples is 11MB). But, they are already disk cached by browsers.
